go-controller/pkg/ovn/ovn.go

```diff
@@ -225,84 +255,193 @@ func getPodNamespacedName(pod *kapi.Pod) string {
 	return util.GetLogicalPortName(pod.Namespace, pod.Name)
 }
 
+func NewControllerManager(ovnClient *util.OVNClientset, identity string, wf *factory.WatchFactory,
+	stopChan chan struct{}, libovsdbOvnNBClient libovsdbclient.Client, libovsdbOvnSBClient libovsdbclient.Client,
+	recorder record.EventRecorder, wg *sync.WaitGroup) *ControllerManager {
+	return &ControllerManager{
+		GenericController: GenericController{
+			client: ovnClient.KubeClient,
+			kube: &kube.Kube{
+				KClient:              ovnClient.KubeClient,
+				EIPClient:            ovnClient.EgressIPClient,
+				EgressFirewallClient: ovnClient.EgressFirewallClient,
+				CloudNetworkClient:   ovnClient.CloudNetworkClient,
+			},
+			watchFactory: wf,
+			recorder:     recorder,
+			nbClient:     libovsdbOvnNBClient,
+			sbClient:     libovsdbOvnSBClient,
+			podRecorder:  metrics.NewPodRecorder(),
+		},
+		defaultWg:       wg,
+		defaultStopChan: stopChan,
+		identity:        identity,
+	}
+}
+
+func (cm *ControllerManager) InitDefaultController(addressSetFactory addressset.AddressSetFactory) (*Controller, error) {
+	defaultNetConf := &cnitypes.NetConf{
+		NetConf: ctypes.NetConf{
+			Name: ovntypes.DefaultNetworkName,
+		},
+		NetCidr:     config.Default.RawClusterSubnets,
+		MTU:         config.Default.MTU,
+		IsSecondary: false,
+	}
+	nadInfo, _ := util.NewNetAttachDefInfo(defaultNetConf)
```

Comment:
I don't think there's any reason for this `util.NewNetAttachDefInfo` to return an error - as such, you should be able to just inline the call:
```go
cm.NewOvnController(util.NewNetAttachDefInfo(defaultNetConf), addressSetFactory)
```

Reply:
there could be error case when we introduce non-default network, as we'd need to validate non-default network netconf.